### PR TITLE
2981/roles mapping unespected behavior

### PR DESCRIPTION
**Pull request by sortizowlh**
Hi team,
This PR checks an invalid behavior that added a logical operator AND or OR without needed.

Closes: #2981

### CODE REVIEW

File: public/components/security/roles-mapping/helpers/rule-editor.helper.ts [hunk 1]
```diff
@@ -11,24 +11,34 @@ export const getJsonFromRule = (internalUserRules, rules, logicalOperator) => {
...
-        OR: usersRulesArray,
-      },
-      {
-        [logicalOperator]: rulesArray,
-      },
-    ];
-  } else {
-    if (usersRulesArray.length) {
-      ruleObject['OR'] = usersRulesArray;
+    if(rulesArray.length == 1){
```
Comment (frankeros) [inline public/components/security/roles-mapping/helpers/rule-editor.helper.ts:17]: this is causing bad behavior because is removing the `userRulesArray`

Comment (frankeros) [reply public/components/security/roles-mapping/helpers/rule-editor.helper.ts:17]: please add some unit test to this function with this uses cases
- Giving userRulesArray NO empty and rulesArray NO empty, should return an object with one `'OR'` array with both arrays as items with all items
- Giving userRulesArray with 1 item  and rulesArray empty, should return the single item
- Giving rulesArray with 1 item  and userRulesArray empty, should return the single item
- Giving rulesArray with 2 o more items  and userRulesArray empty, should return an object with one item equals to the `logicalOperator` (OR or AND) array with all items
 - Giving userRulesArray with 2 o more items  and rulesArray empty, should return an object with one `'OR'` array with all items

**Review by frankeros — CHANGES_REQUESTED** [review]
Tested locally :x: 

-- 

Adding both kind of filters from the UI editor and switch to JSON editor, is removing the user filter
image
image

**Comment by sortizowlh** [comment]
Testing behavior. When adding some code to the JSON editor, the visual editor keeps these changes.

File: public/components/security/roles-mapping/helpers/rule-editor.helper.test.ts [hunk 2]
```diff
@@ -201,4 +201,159 @@ describe('Rule Editor Helper', () => {
       });
     });
   });
-});
+
+  describe('Test role-mapping behavior', () => {
+    describe('Given a non null rules array and non null internal users array ', () => {
```
Comment (gabiwassan) [inline public/components/security/roles-mapping/helpers/rule-editor.helper.test.ts:206]: Here the title says that rules and internal users are null, but they are not. They are correct? Or maybe you can copy this test and create a new one with null values and cover this case 😄 

Great job!

**Review by gabiwassan — CHANGES_REQUESTED** [review]
Great job!
Just a simple comment

**Review by Desvelao — APPROVED** [review]
LGTM!

File: tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "jsx": "react",
```
Comment (Desvelao) [inline tsconfig.json:4]: Is this necessary?

**Comment by Desvelao** [comment]
The change should be added to `CHANGELOG.md` file.

**Review by frankeros — APPROVED** [review]
tested locally :heavy_check_mark: 
Great job :tada:

**Review by pablomarga — APPROVED** [review]
LGTM!

**Comment by Desvelao** [comment]
## Changes
- Updated the branch with the last changes of `4.1-7.10` branch
- Added change into the changelog

**Pull request merged**